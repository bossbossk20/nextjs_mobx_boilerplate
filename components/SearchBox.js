import React from 'react'
import styled from 'styled-components'
import theme from './../theme'

const SearchBoxStyle = styled.div`
  display: flex;
  justify-content: flex-end;
  .searchBoxContainer {
    position: relative;
    display: flex;
    padding:0 0 0 20px;
    margin:0 12px;
    border: 1px solid ${theme().color.darkGray};
    background: ${theme().color.lightBlack};
    /* direction: rtl; */
    height: 34px;
    width: 300px;
    .sendicon-icon-search {
      font-size: 20px;
      position:absolute;
      bottom:0px;
      top:0px;
      left:0px;
      background-color: ${theme().color.lightBlack};
      padding-top: 5px;
      padding-left: 5px;
    }
    .textField {
      color: #FFF;
      border: 0px;
      margin:0;
      padding-right: 30px;
      width: 100%;
      background-color: ${theme().color.lightBlack};
      margin-left: 10px;
    }
    .textField:focus {
        outline: none;
    }
  }
`

const SearchBox = () => (
  <SearchBoxStyle>
    <div className='searchBoxContainer'>
      <input className='textField' placeholder='Search' />
      <i className='sendicon-icon-search' />
    </div>
  </SearchBoxStyle>
)

export default SearchBox
