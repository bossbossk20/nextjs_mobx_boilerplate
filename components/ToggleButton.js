import React, { Component } from 'react'
import styled from 'styled-components'
import theme from './../theme'

const Button = styled.div`
  margin: 0 10px;
  background-color: ${theme().color.black};
  color: ${theme().color.white};
  display: flex;
  justify-content: center;
  border: 2px solid ${theme().color.darkGray};
  width: 180px;
  .side {
    display: flex;
    justify-content: center;
    align-items:center;
    cursor: pointer;
    width: 90px;
    height: 26px;
    font-size: ${theme().font.primary};
  }
`

type Props = {
  handle: (side: string) => void,
  leftTitle: string,
  rightTitle: string
}
export default class ToggleBtn extends Component <Props> {
  state = {
    toggle: 'left'
  }
  handleClick = (side) => {
    this.setState({ toggle: side })
    this.props.handle(side)
  }
  render () {
    const { toggle } = this.state
    const { leftTitle, rightTitle } = this.props
    if (toggle === 'right') {
      return (
        <Button >
          <a className='side'
            style={{backgroundColor: theme().color.lightBlack}}
            onClick={() => this.handleClick('left')}>
            {leftTitle}
          </a>
          <a className='side' style={{backgroundColor: theme().color.primary, color: theme().color.black}}>{rightTitle}</a>
        </Button>
      )
    } else {
      return (
        <Button >
          <a className='side' style={{backgroundColor: theme().color.primary, color: theme().color.black}}>{leftTitle}</a>
          <a
            className='side'
            style={{backgroundColor: theme().color.lightBlack}}
            onClick={() => this.handleClick('right')}>
            {rightTitle}
          </a>
        </Button>
      )
    }
  }
}
