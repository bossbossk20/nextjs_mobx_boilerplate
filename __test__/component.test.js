import Enzyme, { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import 'jest-styled-components'
// import App from '../pages/index.js'

import Input from './../components/Input'
import GoogleMap from './../components/GoogleMap'
import QuickTourButton from './../components/QuickTourButton'
import SearchBox from './../components/SearchBox'
import ToggleButton from './../components/ToggleButton'
import SelectDropdown from './../components/SelectDropdown'
import Button from './../components/Button'

Enzyme.configure({ adapter: new Adapter() })

describe('Components', () => {
  it('Input', () => {
    let tree = renderer.create(<Input />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Google Map', () => {
    let tree = renderer.create(<GoogleMap />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('QuickTourButton', () => {
    let tree = renderer.create(<QuickTourButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('SearchBox', () => {
    let tree = renderer.create(<SearchBox />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ToggleButton', () => {
    let tree = renderer.create(<ToggleButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('SelectDropdown', () => {
    let tree = renderer.create(<SelectDropdown />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Button', () => {
    let tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})


// describe('With Snapshot Testing', () => {
//   it('App shows "Hello world!"', () => {
//     const component = renderer.create(<App />)
//     const tree = component.toJSON()
//     expect(tree).toMatchSnapshot()
//   })
// })