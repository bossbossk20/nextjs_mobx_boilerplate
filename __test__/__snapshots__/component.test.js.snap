// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Button 1`] = `
.c0 {
  background-color: #FFB804;
  color: #000;
  width: 60px;
  height: 60px;
  border-radius: 50%;
  cursor: pointer;
  border: 0px;
}

<button
  className="c0"
/>
`;

exports[`Components Google Map 1`] = `
<div
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;

exports[`Components Input 1`] = `
.c0 {
  color: #FFF;
}

.c0 label {
  color: #FFF;
}

.c0 .MuiInput-inputSingleline-26 {
  color: #FFF;
}

.c0 .MuiFormLabel-focused-12 {
  color: #FFB804 !important;
}

.c0 .MuiInput-inkbar-17:after {
  background-color: #FFF !important;
}

.c0 .MuiInput-underline-23:before {
  background-color: #FFF !important;
}

<div
  className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 c0"
  onBlur={[Function]}
  onFocus={[Function]}
  required={false}
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-inkbar-7 MuiInput-underline-13"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <input
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-9 MuiInput-inputSingleline-16"
      defaultValue={undefined}
      disabled={false}
      id={undefined}
      name={undefined}
      onChange={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder={undefined}
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value={undefined}
    />
  </div>
</div>
`;

exports[`Components QuickTourButton 1`] = `
.c0 {
  background-color: #FFB804;
  color: #000;
  border-radius: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 112px;
  height: 26px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-size: 12px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 12px;
  padding-left: 5px;
}

.c0 .icon {
  margin-left: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFB804;
  background: #000;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  font-size: 13px;
}

.c0 span {
  padding-left: 5px;
  padding-right: 5px;
  padding-top: 5px;
  padding-bottom: 5px;
}

<div
  className="c0"
>
  <span>
    quick tour
  </span>
  <div
    className="icon"
  >
    ?
  </div>
</div>
`;

exports[`Components SearchBox 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c0 .searchBoxContainer {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 0 0 20px;
  margin: 0 12px;
  border: 1px solid #4D4D4D;
  background: #1A1A1A;
  height: 34px;
  width: 300px;
}

.c0 .searchBoxContainer .sendicon-icon-search {
  font-size: 20px;
  position: absolute;
  bottom: 0px;
  top: 0px;
  left: 0px;
  background-color: #1A1A1A;
  padding-top: 5px;
  padding-left: 5px;
}

.c0 .searchBoxContainer .textField {
  color: #FFF;
  border: 0px;
  margin: 0;
  padding-right: 30px;
  width: 100%;
  background-color: #1A1A1A;
  margin-left: 10px;
}

.c0 .searchBoxContainer .textField:focus {
  outline: none;
}

<div
  className="c0"
>
  <div
    className="searchBoxContainer"
  >
    <input
      className="textField"
      placeholder="Search"
    />
    <i
      className="sendicon-icon-search"
    />
  </div>
</div>
`;

exports[`Components SelectDropdown 1`] = `
.c0 {
  padding: 0 20px;
}

.c0 .MuiSelect-selectMenu-7 {
  color: #FFB804;
  font-size: 14px;
  font-weight: bold;
}

.c0 .MuiSelect-icon-9 {
  color: #FFB804;
}

<div
  className="MuiFormControl-root-1"
  onBlur={[Function]}
  onFocus={[Function]}
  required={false}
>
  <div
    className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-inkbar-7 MuiInput-underline-13 c0"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <div
      className="MuiSelect-root-20"
    >
      <div
        aria-haspopup="true"
        aria-owns={null}
        aria-pressed="false"
        className="MuiSelect-select-21 MuiSelect-selectMenu-22 MuiInput-input-9 MuiInput-inputSingleline-16"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={undefined}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        3PL MANAGEMENT
      </div>
      <input
        autoComplete={undefined}
        autoFocus={undefined}
        defaultValue={undefined}
        id={undefined}
        name={undefined}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-25 MuiSelect-icon-24"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Components ToggleButton 1`] = `
.c0 {
  margin: 0 10px;
  background-color: #000;
  color: #FFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 2px solid #4D4D4D;
  width: 180px;
}

.c0 .side {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  width: 90px;
  height: 26px;
  font-size: 12px;
}

<div
  className="c0"
>
  <a
    className="side"
    style={
      Object {
        "backgroundColor": "#FFB804",
        "color": "#000",
      }
    }
  />
  <a
    className="side"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#1A1A1A",
      }
    }
  />
</div>
`;
